src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java

```diff
@@ -102,16 +105,19 @@ protected int calculateMaxCraftingCount(@NotNull final ItemStack outputStack, @N
             if (ingredientInputCount > ingredient.getMaxStackSize())
             {
                 //Recalculate the crafting limit using the maxstacksize of the ingredient.
-                craftingCount =
-                  (int) Math.ceil(Math.max(ingredient.getMaxStackSize(), ItemStackUtils.getSize(storage.getPrimaryOutput())) / ItemStackUtils.getSize(storage.getPrimaryOutput()));
+                craftingCount = Math.max(ingredient.getMaxStackSize(), ItemStackUtils.getSize(storage.getPrimaryOutput())) / ItemStackUtils.getSize(storage.getPrimaryOutput());
             }
         }
 
         return craftingCount;
     }
 
     @Nullable
-    protected List<IToken<?>> createRequestsForRecipe(@NotNull final IRequestManager manager, @NotNull final AbstractBuildingWorker building, final ItemStack requestStack, @NotNull final IRecipeStorage storage)
+    protected List<IToken<?>> createRequestsForRecipe(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)